### PR TITLE
[tasks] Include Names for `include_tasks`

### PR DESCRIPTION
**Pull request by the-real-cphillips**
Currently, the output when running this role is confusing.
It isn't clear what exactly is being skipped for each `include_tasks` play.

This was initially confusing until I dug into the actual `tasks/main.yml`
This PR adds a `name:` parameter to each `include_tasks` within the `tasks/main.yml`

While a small nit-picky PR, It makes the output clearer and easier to explain to others, without having to dive into code.

### CODE REVIEW

**Comment by KSerrania** [comment]
Hello @the-real-cphillips,

Thanks for the PR! We released a new major version of the role this Wednesday, so the task list got modified. Would you mind rebasing your PR and adding the `name:` parameter to the new and modified tasks?

**Comment by the-real-cphillips** [comment]
> Hello @the-real-cphillips,
> 
> Thanks for the PR! We released a new major version of the role this Wednesday, so the task list got modified. Would you mind rebasing your PR and adding the `name:` parameter to the new and modified tasks?

Sure thing! I'll get on it now :)

File: tasks/main.yml [hunk 1]
```diff
@@ -1,33 +1,45 @@
...
-- include_tasks: agent5-linux.yml
-  when: datadog_agent_major_version|int == 5 and ansible_os_family != "Windows"
+- name: Agent5 Install Tasks (Linux)
+  include_tasks: agent5-linux.yml
+  when: datadog_agent5 and ansible_os_family != "Windows"
 
-- include_tasks: agent-linux.yml
-  when: datadog_agent_major_version|int > 5 and ansible_os_family != "Windows"
+- name: Agent5 Install Tasks (Linux)
+  include_tasks: agent6-linux.yml
```
Comment (KSerrania) [inline tasks/main.yml]:
```suggestion
  include_tasks: agent-linux.yml
```

File: tasks/main.yml [hunk 2]
```diff
@@ -1,33 +1,45 @@
...
+  when: not datadog_agent5 and ansible_os_family != "Windows"
```
Comment (KSerrania) [inline tasks/main.yml]:
```suggestion
  when: datadog_agent_major_version|int > 5 and ansible_os_family != "Windows"
```

File: tasks/main.yml [hunk 3]
```diff
@@ -1,33 +1,45 @@
...
-- include_tasks: pkg-windows.yml
+- name: Windows Install Tasks
+  include_tasks: pkg-windows.yml
   when: ansible_os_family == "Windows"
 
-- include_tasks: agent5-linux.yml
-  when: datadog_agent_major_version|int == 5 and ansible_os_family != "Windows"
+- name: Agent5 Install Tasks (Linux)
+  include_tasks: agent5-linux.yml
+  when: datadog_agent5 and ansible_os_family != "Windows"
```
Comment (KSerrania) [inline tasks/main.yml]:
```suggestion
  when: datadog_agent_major_version|int == 5 and ansible_os_family != "Windows"
```

File: tasks/main.yml [hunk 4]
```diff
@@ -1,33 +1,45 @@
...
-- include_tasks: agent-linux.yml
-  when: datadog_agent_major_version|int > 5 and ansible_os_family != "Windows"
+- name: Agent5 Install Tasks (Linux)
+  include_tasks: agent6-linux.yml
+  when: not datadog_agent5 and ansible_os_family != "Windows"
 
-- include_tasks: agent-win.yml
-  when: datadog_agent_major_version|int > 5 and ansible_os_family == "Windows"
+- name: Agent6 Install Tasks (Windows)
+  include_tasks: agent6-win.yml
```
Comment (KSerrania) [inline tasks/main.yml]:
```suggestion
  include_tasks: agent-win.yml
```

File: tasks/main.yml [hunk 5]
```diff
@@ -1,33 +1,45 @@
...
+  when: not datadog_agent5 and ansible_os_family == "Windows"
```
Comment (KSerrania) [inline tasks/main.yml]:
```suggestion
  when: datadog_agent_major_version|int > 5 and ansible_os_family == "Windows"
```

File: tasks/main.yml [hunk 6]
```diff
@@ -1,33 +1,45 @@
...
 
-- include_tasks: agent5-linux.yml
-  when: datadog_agent_major_version|int == 5 and ansible_os_family != "Windows"
+- name: Agent5 Install Tasks (Linux)
+  include_tasks: agent5-linux.yml
+  when: datadog_agent5 and ansible_os_family != "Windows"
 
-- include_tasks: agent-linux.yml
-  when: datadog_agent_major_version|int > 5 and ansible_os_family != "Windows"
+- name: Agent5 Install Tasks (Linux)
```
Comment (KSerrania) [inline tasks/main.yml]:
```suggestion
- name: Linux Install Tasks
```

File: tasks/main.yml [hunk 7]
```diff
@@ -1,33 +1,45 @@
...
   when: ansible_os_family == "Suse"
 
-- include_tasks: pkg-windows.yml
+- name: Windows Install Tasks
+  include_tasks: pkg-windows.yml
   when: ansible_os_family == "Windows"
 
-- include_tasks: agent5-linux.yml
-  when: datadog_agent_major_version|int == 5 and ansible_os_family != "Windows"
+- name: Agent5 Install Tasks (Linux)
```
Comment (KSerrania) [inline tasks/main.yml]:
```suggestion
- name: Linux Install Tasks (Agent 5)
```

**Review by KSerrania — CHANGES_REQUESTED** [review]
Thanks! I'm adding some suggested changes to make the role work as expected, the merge reverted some changes that were done in the major version bump (mainly some filename changes, and datadog_agent5 got replaced by datadog_agent_major_version).

Once these are added the PR should be good to go!

File: tasks/main.yml (hunk 1 above)
Comment (KSerrania) [reply tasks/main.yml]: @the-real-cphillips I think you forgot or misclicked on this one, the main (non-Agent 5) linux install file is named [agent-linux.yml](https://github.com/the-real-cphillips/ansible-datadog/blob/name_parameter_for_includes/tasks/agent-linux.yml) now

Comment (the-real-cphillips) [reply tasks/main.yml]: you are correct 🙈 
I just fixed in latest commit

**Review by KSerrania — APPROVED** [review]
Thanks! 
I'm going to merge this PR and release a new version on Monday.

As you mentioned in your commits, we have a few linting errors, but they're tackled by #236 so everything's good here!

**Comment by the-real-cphillips** [comment]
> Thanks!
> I'm going to merge this PR and release a new version on Monday.
> 
> As you mentioned in your commits, we have a few linting errors, but they're tackled by #236 so everything's good here!

Awesome thanks!

**Comment by the-real-cphillips** [comment]
@KSerrania probably unrelated here, but has there been any talk about including molecule for testing the role to ensure it's idempotent and works as expected? 

I plausibly would be willing to undertake such a thing if there's desired need/want for it.

**Pull request merged**